metadata-ingestion/src/datahub/utilities/threaded_iterator_executor.py

```diff
@@ -18,8 +26,13 @@ def process(
         worker_func: Callable[..., Iterable[T]],
         args_list: Iterable[Tuple[Any, ...]],
         max_workers: int,
-    ) -> Generator[T, None, None]:
-        out_q: queue.Queue[T] = queue.Queue()
+        max_backpressure: Optional[int] = None,
+    ) -> Iterator[T]:
+        if max_backpressure is None:
+            max_backpressure = 10 * max_workers
+        assert max_backpressure >= max_workers
```

Comment:
I understand this assertion is to avoid wasting resources (if queue size is higher than max_workers then probably it is pointless to run that many), but process would still succeed, right?
How about adding some helpful explanation to this assertion, considering not meeting this assertion would cause an Exception visible to (possibly) non-tech-savvy users trying to configure ingestion via the UI?

Reply:
+1

Reply:
The only setting that is user-facing is max_workers. The max_backpressure arg is there only as an escape hatch in case we need it, but is not used actively by anyone. Given that a low max_backpressure value would cause max_workers to not really be respected, I think it still merits an exception.

In general - code should fail loudly, not quietly. Our ingestion sources are responsible for converting exceptions -> structured report messages